Oss to ee (#143)
kestra-io/kestra.io: docs/administrator-guide/configuration/README.md

---
# Configuration

Kestra offers a lot of configuration options and customization.

How to pass those configuration options depends on how you deploy Kestra. Check the [deployment](/docs/administrator-guide/deployment/README.md) section related to you deployment mode for more information.

## Kestra internal components configuration 

Kestra has three internal components that must be configured:
- The Internal Storage.
- The Queue.
- The Repository.

### Internal storage configuration

> Kestra supports multiple internal storage types, the default being the local storage that will store data inside a local folder on the host filesystem. **Only suitable for local testing** as it didn't provide resiliency nor redundancy.

Reviewer: ```suggestion
Kestra supports multiple internal storage types, the default being the local storage that will store data inside a local folder on the host filesystem. **Only suitable for local testing** as it doesn't provide resiliency or redundancy.
```

> To choose another storage type, you will need to configure the `kestra.storage.type` option, be sure to download the corresponding plugins first. The following example configure [Google Cloud Storage](./storage/README.md#gcs) for the internal storage.

Reviewer: ```suggestion
To choose another storage type, you will need to configure the `kestra.storage.type` option, be sure to download the corresponding plugins first. The following example configures [Google Cloud Storage](./storage/README.md#gcs) for internal storage.
```

Author: I use "the" as it's "the internal storage" of Kestra, one of its component.

```yaml
kestra:
  storage:
    type: gcs
```

> Details about Storage configuration can be found [here](./storage).

Reviewer: ```suggestion
Details about storage configuration can be found [here](./storage).
```

### Queue configuration 

> Kestra supports multiple queue types, the default being an in-memory queue that is **only suitable for local testing** as it didn't provide any resiliency nor scalability. In-memory queue must be used with the in-memory repository.

Reviewer: ```suggestion
Kestra supports multiple queue types, the default being an in-memory queue that is **only suitable for local testing** as it doesn't provide any resiliency or scalability. The in-memory queue must be used with the in-memory repository.
```

> The queue type must be compatible with the repository type, not all combinations are possible.

Reviewer: ```suggestion
The queue type must be compatible with the repository type. Not all combinations are possible.
```

The following queue types are available:
- In-memory that must be used with the in-memory repository.

> - Database that must be used with the database repository, it currently supports H2, MySQL or PostgreSQL as a database.

Reviewer: ```suggestion
- Database that must be used with the database repository. It currently supports H2, MySQL, and PostgreSQL as a database.
```

> - Kafka that must be used with the Elasticsearch repository, those are **only available inside the Enterprise Edition**.

Reviewer: ```suggestion
- Kafka that must be used with the Elasticsearch repository. Those are **only available in the Enterprise Edition**.
```

> To enable the PostgreSQL database Queue, you need to add the following to your configuration files:

Reviewer: ```suggestion
To enable the PostgreSQL database queue, you need to add the following to your configuration files:
```

```yaml
kestra:
  queue:
    type: postgres
```

Details about the database configuration can be found [here](./databases/) and about the Kafka configuration [here](./kafka).

### Repository configuration

> Kestra supports multiple repository types, the default being an in-memory repository that is **only suitable for local testing** as it didn't provide any resiliency nor scalability and didn't implement all functionalities. In-memory repository must be used with the in-memory queue.

Reviewer: ```suggestion
Kestra supports multiple repository types, the default being an in-memory repository that is **only suitable for local testing** as it doesn't provide any resiliency or scalability and didn't implement all functionalities. The in-memory repository must be used with the in-memory queue.
```

> The repository type must be compatible with the queue type, not all combinations are possible.

Reviewer: Same updates as the above suggestions.

The following repository types are available:
- In-memory that must be used with the in-memory queue.
- Database that must be used with the database queue, it currently supports H2, MySQL or PostgreSQL as a database.
- Elasticsearch that must be used with the Kafka repository, those are **only available inside the Enterprise Edition**.

To enable the PostgreSQL database repository, you need to add the following to your configuration files:
```yaml
kestra:
  repository:
    type: postgres
```

Details about the database configuration can be found [here](./databases/) and about the Elasticsearch configuration [here](./elasticsearch).

## Other Kestra configuration

Other Kestra configuration options can be found [here](./others).

## Micronaut configuration

> As Kestra is a Java based application built with Micronaut, you can configure any Micronaut configuration options, more details [here](./micronaut).

Reviewer: ```suggestion
As Kestra is a Java-based application built with Micronaut, you can configure any Micronaut configuration options. More details [here](./micronaut).
```
